client/views/components/generic/date-utility/DateUtility.jsx

```diff
@@ -0,0 +1,61 @@
+import React, {PureComponent } from 'react';
+import { string, object  } from 'prop-types';
+import { BaseEntityQuery } from 'utils/genny';
+import distanceInWords from 'date-fns/distance_in_words';
+
+class DateUtility extends PureComponent {
+    static propTypes = {
+        attribute: string,
+        style: object,
+        field: string,
+        code: string,
+        formatType: string
+    }
+
+    getFromNowInWords = (code, attribute, field) => { 
+      /* if BECODE and ATTRIBUTE is supplied as props */
+      if(code && attribute){ 
+        const attributevalues = BaseEntityQuery.getBaseEntityAttribute(code, attribute);
+        const dateValue = attributevalues.valueDate;
+        if(dateValue) { 
+          const timeAgo =  distanceInWords(dateValue, Date.now());
+          return timeAgo;
+        }
+      }
+      /* if BECODE and field is supplied as props */
+      if(code && field) { 
+        const fieldValue = BaseEntityQuery.getBaseEntityField(code , field);  
+        if(fieldValue) { 
+          const timeAgo =  distanceInWords(fieldValue, Date.now());
+          return timeAgo;
+        }
+      }
+      return null;
+    }
+
+    /* choose the type of formatting to apply and call the function accordingly */
+    formatSelection (code, attribute, field, formatType){ 
+      switch (formatType) { 
```

Comment:
what's the purpose of the switch if it calls the same function

Reply:
I made this generic so that there could be lot of date formatting 
for example tima-ago-in-words, time-difference-in-words etc etc.